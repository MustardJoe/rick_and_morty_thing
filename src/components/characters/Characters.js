import React from 'react';
import PropTypes from 'prop-types';
import Character from './Character';
import styles from './Characters.css';

function Characters({ characters }) {
  //eslint-desable-next-line
  const characterList = characters.map((character, i) => ( //eslint-disable-line react/prop-types

    <li key={i}>
      <Character character={character} />
    </li>
  ));

  return (
    <ul className={styles.Characters}>
      {characterList}
    </ul>
  );
}

Characters.propTypes = {
  characters: PropTypes.array.isRequired,
};

export default Characters;
