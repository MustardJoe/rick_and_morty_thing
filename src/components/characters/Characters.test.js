import React from 'react';
import { shallow } from 'enzyme';
import Characters from './Characters';

describe('Characters component test', () => {
  it('renders the Characters comp', () => {
    const characters = [
      {
        name: 'morthy',
        status: 'dead',
        species: 'five dead',
        image: '/image',
      },
      {
        name: 'barnman the bat barn man',
        status: 'berry much alive',
        species: 'the regular',
        image: '/fivepics',
      }
    ];

    const wrapper = shallow(<Characters characters={characters} />);
    expect(wrapper).toMatchSnapshot();
  });
});
