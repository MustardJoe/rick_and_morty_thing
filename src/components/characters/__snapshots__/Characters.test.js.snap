// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Characters component test renders the Characters comp 1`] = `
<ul
  className="Characters"
>
  <li
    key="0"
  >
    <Character
      character={
        Object {
          "image": "/image",
          "name": "morthy",
          "species": "five dead",
          "status": "dead",
        }
      }
    />
  </li>
  <li
    key="1"
  >
    <Character
      character={
        Object {
          "image": "/fivepics",
          "name": "barnman the bat barn man",
          "species": "the regular",
          "status": "berry much alive",
        }
      }
    />
  </li>
</ul>
`;
